File: leetcode/2997.minimum-number-of-operations-to-make-array-xor-equal-to-k/solution.go
package main

import "fmt"

func minOperations(nums []int, k int) int {
	bits_diff := 0

	for i := 0; i < len(nums); i++ {
		bits_diff ^= nums[i]
	}

	count := 0
	bits_diff ^= k

	for bits_diff != 0 {
		count += (bits_diff & 1)
		bits_diff >>= 1
	}

	return count
}

func main() {
	arr := []int{2, 1, 3, 4}
	k := 1
	expected := 2

	actual := minOperations(arr, k)

	if actual != expected {
		fmt.Printf("Expected %d but found %d", expected, actual)
		return
	}

	fmt.Printf("Correct %d", actual)
}

File: leetcode/2997.minimum-number-of-operations-to-make-array-xor-equal-to-k/solution.c
#include <assert.h>
#include <stdio.h>

int min_operations(int *nums, int nums_size, int k)
{
    int bits_diff = 0;

    for (int i = 0; i < nums_size; ++i) {
        bits_diff ^= nums[i];
    }

    bits_diff ^= k;
    int count = 0;

    while (bits_diff) {
        count += bits_diff & 1;
        bits_diff >>= 1;
    }

    return count;
}

int main(void)
{
    int nums[] = {2, 1, 3, 4};
    int k = 1;
    int expected = 2;

    int actual = min_operations(nums, 4, k);

    assert(actual == expected);
    printf("Correct: %d", actual);

    return 0;
}
